### PR TITLE
[Fix #3455] Add new `Rails/DynamicFindBy` cop

### LINKED ISSUE #3455: find_by_* cop
I'd like to suggest a new cop, which will look for instances of the old rails dynamic finders (ie: `find_by_id`) and prefer the non-dynamic version: (`find_by(id: ...)`)

Edit: whoever implements this, be careful of instances such as `find_by_sql`, which is a legit method. Not sure if there are others.

### PR DESCRIPTION
**Pull request by pocke**
Resolve #3455

Feature
----

This cop checks dynamic `find_by_*` method.

```ruby
# Bad code
User.find_by_email(email)
User.find_by_name!(name)
User.find_by_email_and_name(email, name)

# Good code
User.find_by(email: email)
User.find_by!(name: name)
User.find_by(email: email, name: name)
```

Note
-----

When arguments size and column name size are different,
the cop registers an offence.
However, doesn't auto-correct.

e.g.

```ruby
# It's a bad code, howver, can't auto correct.
User.find_by_name_and_email(name)
```

-----------------

Before submitting the PR make sure the following are checked:

* [x] Wrote [good commit messages][1].
* [x] Commit message starts with `[Fix #issue-number]` (if the related issue exists).
* [x] Used the same coding conventions as the rest of the project.
* [x] Feature branch is up-to-date with `master` (if not - rebase it).
* [x] Squashed related commits together.
* [x] Added tests.
* [x] Added an entry to the [Changelog](../blob/master/CHANGELOG.md) if the new code introduces user-observable changes. See [changelog entry format](../blob/master/CONTRIBUTING.md#changelog-entry-format).
* [x] All tests are passing.
* [x] The new code doesn't generate RuboCop offenses.
* [x] The PR relates to *only* one subject with a clear title
  and description in grammatically correct, complete sentences.

[1]: http://tbaggery.com/2008/04/19/a-note-about-git-commit-messages.html

### CODE REVIEW

File: config/enabled.yml [hunk 1]
```diff
@@ -1435,6 +1435,10 @@ Rails/Delegate:
   Description: 'Prefer delegate method for delegations.'
   Enabled: true
 
+Rails/DynamicFindBy:
+  Description: 'Use `find_by` instead of dynamic `find_by_*`'
```
Comment (bbatsov) [inline config/enabled.yml]: This sentence should end with a `.`.

**Comment by bbatsov** [comment]
Apart from my tiny remark, the new cop looks good.

**Comment by pocke** [comment]
Fixed.
Thanks for the review.

**Comment by bbatsov** [comment]
You'll also have to rebase on top of the current `master`.

**Comment by bbatsov** [comment]
I hate our changelog... Another merge conflict there...

**Comment by pocke** [comment]
np. rebased.

We need better management way of CHANGELOG. But I don't have any idea :cry:

**Comment by bbatsov** [comment]
> We need better management way of CHANGELOG. But I don't have any idea 😢

It's a tough nut to crack - have been wondering about this for a while. Some people generate the changelog from the commit messages, but that's also problematic.

**Pull request merged**